oat-sa/extension-tao-testqti
# Fix/tca 384/disable tools via lti

Overridden test category removal by fuzzy matching.
 
Related to : https://oat-sa.atlassian.net/browse/TCA-384
 
Test authors can specify any categories they wish for the test items. f.e. `x-tao-option-line-reader`. Such added category enables Line Reader plugin. Correct option key is in camelCase (`x-tao-option-lineReader`), but frontend is fuzzy matching the strings and the previous version is valid as well. https://github.com/oat-sa/tao-test-runner-qti-fe/blob/0a25a0c8b2895426514ef66037b5ddceb8124836/src/helpers/map.js#L233
This PR modifies how test options are overriden in backend, categories to be removed are found by fuzzy matching as well.
  
#### How to test
 
 - create a test with category `x-tao-option-line-reader` assigned to the section, publish a delivery.
 - launch this test via LTI and verify that line reader tool is enabled
 - launch same test via LTI, with added custom parameter `x_tao_tools={"lineReader":false}`. Verify that line reader tool is disabled
 
```
./vendor/bin/phpunit taoQtiTest/test/unit/models/classes/runner/map/QtiRunnerMapTest.php
```
